sc/social/like/utils.py

```diff
@@ -128,3 +128,11 @@ def get_valid_objects(brains):
             logger.warn(msg.format(b.getPath()))
             continue
         yield obj
+
+
+def get_path_to_virtual_path(request, item):
```

Comment:
I think there's no need to create this function; just place the code inside the `assign_canonical_url` subscriber, as this is the only place where it's called..